cfme/infrastructure/pxe.py

```diff
@@ -349,14 +509,23 @@ def create(self, cancel=False):
             cancel (boolean): Whether to cancel out of the creation.  The cancel is done
                 after all the information present in the CT has been filled in the UI.
         """
-        navigate_to(self, 'Add')
-        fill(template_properties_form, self._form_mapping(True, **self.__dict__))
-        self._submit(cancel, form_buttons.add)
-        if not cancel:
-            flash.assert_message_match('Customization Template "{}" was saved'.format(self.name))
+        view = navigate_to(self, 'Add')
+
+        view.fill({'name': self.name,
+                   'description': self.description,
+                   'image_type': self.image_type,
+                   'type': self.script_type,
+                   'script': self.script_data})
+        main_view = self.create_view(PXECustomizationTemplatesView)
+
+        if cancel:
+            view.cancel.click()
+            msg = 'Add of new Customization Template was cancelled by the user'
+            main_view.flash.assert_success_message(msg)
         else:
-            flash.assert_message_match(
-                'Add of new Customization Template was cancelled by the user')
+            view.add.click()
+            msg = 'Customization Template "{}" was saved'.format(self.name)
+            main_view.flash.assert_success_message(msg)
```

Comment:
This line is twice here, you can move it outside of the if blocks for common execution and also move the main_view instantiation here as well to be together.